src/Equinox.Core/Cache.fs

```diff
@@ -34,19 +34,22 @@ type private CacheEntry<'state>(initialToken: StreamToken, initialState: 'state,
     /// Coordinates having a max of one in-flight request across all staleness-tolerant loads at all times
     // Follows high level flow of AsyncCacheCell.Await - read the comments there, and the AsyncCacheCell tests first!
     member x.ReadThrough(maxAge: TimeSpan, isStale, load: Func<_, _>) = task {
+        let act = System.Diagnostics.Activity.Current
+        let setCacheHit hit = if act <> null then act.SetTag(Tags.cache_hit, hit) |> ignore
         let cacheEntryValidityCheckTimestamp = System.Diagnostics.Stopwatch.GetTimestamp()
-        let isWithinMaxAge cachedValueTimestamp = Stopwatch.TicksToSeconds(cacheEntryValidityCheckTimestamp - cachedValueTimestamp) <= maxAge.TotalSeconds
+        let age timestamp = Stopwatch.TicksToSeconds(cacheEntryValidityCheckTimestamp - timestamp)
+        if act <> null then act.SetTag(Tags.cache_age, 1000. * age verifiedTimestamp) |> ignore
```

Comment:
doesn't really matter but arguably this should be inside fetchStateConsistently
and probably calc age into a temporary (once) inline there